### PR TITLE
Update map -> map(*) and empty() -> empty-sequence() according to spec.

### PR DESCRIPTION
**Pull request by dizzzz**
### Description:

Update `map -> map(*)` and `empty() -> empty-sequence()` according to spec.
and: `element(*)`, `function(*)` and `array(*)`

### Reference:

https://www.w3.org/TR/xquery-31/#doc-xquery31-SequenceType
https://www.w3.org/TR/xquery-31/#doc-xquery31-MapTest

### Type of tests:

- existing tets suite --> OK
- pending: additional test to show progression

@wolfgangmm is this a correct approach?

### CODE REVIEW

File: src/org/exist/xquery/value/Type.java [hunk 1]
```diff
@@ -189,7 +189,7 @@
         //TODO use parentheses after the nodes name  ?
         defineBuiltInType(NODE, "node()");
         defineBuiltInType(ITEM, "item()");
-        defineBuiltInType(EMPTY, "empty()");
+        defineBuiltInType(EMPTY, "empty-sequence()","empty");
```
Comment (shabanovd) [inline src/org/exist/xquery/value/Type.java]: it must be `empty()`

**Comment by dizzzz** [comment]
Nono according to the spec, as i wrote

**Comment by shabanovd** [comment]
@dizzzz please, stop a bit and think before reply ... you're giving alternative name `empty` but it must be `empty()`

File: src/org/exist/xquery/value/Type.java (hunk 1 above)
Comment (dizzzz) [reply src/org/exist/xquery/value/Type.java]: My mobile app did not su the correct comment' you are right, I'll update.
Te question remains... is this the correct change? is it an external interface change which means we need to hold it till eXist-db 4.0.0 ?

Comment (shabanovd) [reply src/org/exist/xquery/value/Type.java]: no needs to wait till 4.x as it support both, remove `empty()` or `map` will require 4.x

**Comment by dizzzz** [comment]
@adamretter I was able to think of some tests..... things are not too difficult sometimes.
I found more examples where exist-db did not follow the spec. backward compliance is present.

**Comment by joewiz** [comment]
@dizzzz Regarding your comment:

> I found more examples where exist-db did not follow the spec. backward compliance is present.

Can you think of any way to document which syntax is being supported for backward compatibility? This would be very helpful to be able to consult. Also, ideally, we should mark or declare such syntax that isn't compliant with the spec as deprecated - and then drop support for it in an upcoming major version of eXist.

**Comment by dizzzz** [comment]
Marking stuff deprecated is not possible in the code; indeed I forgot about the array test :)

All comes from the 3.1 specs, I searched literally for (*) in there. Also element is marked this way.

I’ll updated the pr on your request ....

**Comment by dizzzz** [comment]
In the end, this code looks only to be used for the xqdoc code; I actually found the issues with the fundocs app (Mongrel development)

**Comment by adamretter** [comment]
@dizzzz @joewiz I am not sure about `element(*)`. I see both `element()` and `element(*)` being used throughout the spec. Unless anyone can suggest a good reason otherwise, I think we should stay with `element()`

**Comment by dizzzz** [comment]
I reverted it to element() without the * in the last commit

**Comment by dizzzz** [comment]
The title here https://github.com/eXist-db/exist/pull/1680/commits/3a44cc8de2b38d01c18efb8bb79eed91fa462d70 is wrong, sorry

**Comment by joewiz** [comment]
Indeed, while this PR contains tests showing the correct syntax works, it didn't include tests demonstrating backward compatibility is maintained.

**Pull request merged**